### DISCUSSION TITLE
"process is undefined" when used as a function in a store

### DISCUSSION THREAD

**Discussion by straurob** [quasarframework/quasar#13386] [General - Components / Directives / etc] [answered]
The following snippet represents a Pinia store in  my Vue 3 / Quasar 2 application. This store uses the environment variable `VUE_APP_BACKEND_API_URL` which shall be read from either the `window` object or `process.env`, depending on which is available.

The `window.appConfig` object will be created before my nginx serves the application, whereas `process.env` shall be used during local development.

However I don't understand why the first variant is wokring but the second is not. Using the `getEnv` function always results in a `Uncaught (in promise) ReferenceError: process is not defined` error.

```js
import { defineStore } from 'pinia';

function getEnv(name) {
  return window?.appConfig?.[name] || process.env[name];
}

// 1. this is working
const backendApiUrl = window?.appConfig?.VUE_APP_BACKEND_API_URL || process.env.VUE_APP_BACKEND_API_URL;

// 2. this is NOT working
const backendApiUrl = getEnv('VUE_APP_BACKEND_API_URL');

export const useAppConfigStore = defineStore('appConfig', {
  state: () => ({
    authorizationUrl: new URL(
      '/oauth2/authorization/keycloak',
      backendApiUrl,
    ).toString(),
    logoutUrl: new URL('/logout', backendApiUrl).toString(),
    backendApiUrl: new URL(backendApiUrl).toString(),
  }),
});
```

**Comment by metalsadman** [comment, marked as answer]
you can't refer to `process.env` properties dynamically. https://github.com/quasarframework/quasar/issues/8361
